### PR TITLE
Add implementation with React TypeScript

### PR DESCRIPTION
**Pull request by satrio-pamungkas**
### What's New : 
- Add functionality and style similar to vanilla js project
- Refactor some logic to adapt with React TypeScript environment
- Add shuffle algorithm using Fisher–Yates algorithm
- Add starter project using Vite instead create-react-app
- Add guidelines for setup and installation instruction

### Changes :
- Fix issue when two or more times same card clicked it just reveal other match card

### CODE REVIEW

File: TypeScript/TypeScript-React/src/App.tsx [hunk 1]
```diff
@@ -1,15 +1,83 @@
...
+      setClickable(true);
+      console.log('It matched !');
+
+    } else {
+      setOpenCards([]);
+      setClickable(true);
+    }
+  }
+
+  const cardClicked = (idx: number) => {
```
Comment (GeraAlcantara) [inline TypeScript/TypeScript-React/src/App.tsx:35]: If I click the card two time it open the one is matching

**Review by GeraAlcantara — CHANGES_REQUESTED** [review]

File: TypeScript/TypeScript-React/src/App.tsx (hunk 1 above)
Comment (GeraAlcantara) [reply TypeScript/TypeScript-React/src/App.tsx:35]: try to click the card two time and you will see that auto open the matching card

Comment (GeraAlcantara) [reply TypeScript/TypeScript-React/src/App.tsx:35]: I thing is 3 clicks to the same card when is open that open the hidden one resulting in match

Comment (satrio-pamungkas) [reply TypeScript/TypeScript-React/src/App.tsx:35]: Yeah you're right, missing conditional to check clicked card index similarity. 
I'm fix it with new commits. Please have a look

[Fix Commit](https://github.com/GeraAlcantara/cards-memory-game/pull/24/commits/2076d7cb4af84d071035f58bc799af2ca25eca43)

Comment (GeraAlcantara) [reply TypeScript/TypeScript-React/src/App.tsx:35]: Look great Thank you!  🌶️

**Review by GeraAlcantara — APPROVED** [review]

**Pull request merged**